### ISSUE TITLE
 Caused by java.lang.IllegalArgumentException No such service ComponentInfo{com.bloomer.alaWad3k/com.onesignal.FocusDelaySyncJobService}

### ISSUE THREAD

**Issue by mahmoudElfeel67** [OneSignal/OneSignal-Android-SDK#1435]
**Description:**
i am having a crash with a couple of users with
 Caused by java.lang.IllegalArgumentException
No such service ComponentInfo{com.bloomer.alaWad3k/com.onesignal.FocusDelaySyncJobService}

(your description here)

**Environment**
1. What version of the Android SDK are you using?
2. How did you add the SDK to your project (eg. maven)

**Steps to Reproduce Issue:**

  1. Add the OneSignal SDK to your project 4.5.1
  2. Place your App ID in the application's gradle file a4a5c56d-e19b-4b2a-a0af-af653115977d
  3. Attempt to receive a push notification 

i cant reproduce

**Anything else:**

(crash stacktraces, as well as any other information here)
```java
Fatal Exception: java.lang.RuntimeException: Unable to pause activity {com.bloomer.alaWad3k/com.bloomer.alaWad3k.kot.ui.activty.AuthActivity}: java.lang.IllegalArgumentException: 
No such service ComponentInfo{com.bloomer.alaWad3k/com.onesignal.FocusDelaySyncJobService}
       at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:3961)
       at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3927)
       at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3901)
       at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:3875)
       at android.app.ActivityThread.-wrap15(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1625)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6545)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:857)
```

**Comment by nan-li** [comment]
Hi @mahmoudElfeel67 and @sanduluca, can you provide more information?
1. Android versions and device models that are seeing the issue? 
2. Percentage of users affected by this crash?
3. Is this crash reproducible? (@sanduluca)

**Comment by sanduluca** [comment]
I dont have the logs for this exception anymore. So I can't provide any details.
I also could not reproduce that.

I got an ANR this time

**Device**
Brand:samsung
Model:Galaxy A03s
Orientation: Unknown
RAM free: Unknown
Disk free: Unknown

**Operating System**
Version:Android 11
Orientation: Unknown
Rooted:No

`
main (native): tid=1 systid=15738
#00 pc 0x86f8c libc.so (syscall + 28)
#01 pc 0x1b27a4 libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*) + 148)
#02 pc 0x67e9d0 libart.so (art::GoToRunnable(art::Thread*) + 480)
#03 pc 0x67e7ac libart.so (art::JniMethodEnd(unsigned int, art::Thread*) + 28)
       at android.os.BinderProxy.transactNative(Native method)
       at android.os.BinderProxy.transact(BinderProxy.java:605)
       at android.app.job.IJobScheduler$Stub$Proxy.getAllPendingJobs(IJobScheduler.java:425)
       at android.app.JobSchedulerImpl.getAllPendingJobs(JobSchedulerImpl.java:90)
       at com.onesignal.OSBackgroundSync.isJobIdRunning(OSBackgroundSync.java:106)
       at com.onesignal.OSBackgroundSync.scheduleSyncServiceAsJob(OSBackgroundSync.java:118)
       at com.onesignal.OSBackgroundSync.scheduleBackgroundSyncTask(OSBackgroundSync.java:90)
       at com.onesignal.OSFocusDelaySync.scheduleSyncTask(OSFocusDelaySync.java:75)
       at com.onesignal.ActivityLifecycleHandler.handleLostFocus(ActivityLifecycleHandler.java:173)
       at com.onesignal.ActivityLifecycleHandler.onActivityPaused(ActivityLifecycleHandler.java:95)
       at com.onesignal.ActivityLifecycleListener.onActivityPaused(ActivityLifecycleListener.java:97)
       at android.app.Application.dispatchActivityPaused(Application.java:485)
       at android.app.Activity.dispatchActivityPaused(Activity.java:1449)
       at android.app.Activity.onPause(Activity.java:2387)
       at androidx.fragment.app.FragmentActivity.onPause(FragmentActivity.java:372)
       at com.facebook.react.ReactActivity.onPause(ReactActivity.java:50)
       at android.app.Activity.performPause(Activity.java:8392)
       at android.app.Instrumentation.callActivityOnPause(Instrumentation.java:1510)
       at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:5237)
       at android.app.ActivityThread.performPauseActivity(ActivityThread.java:5198)
       at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:5149)
       at android.app.servertransaction.PauseActivityItem.execute(PauseActivityItem.java:46)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2317)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:246)
       at android.app.ActivityThread.main(ActivityThread.java:8595)
       at java.lang.reflect.Method.invoke(Native method)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
`

**Comment by jkasten2** [comment]
@sanduluca The issue you reported is different, this is an ANR (main thread stall) instead of a crash like the other reports.

The issue you are looking for is #1315, please follow up there with your details.

**Comment by nan-li** [comment]
Thanks for reporting these details. The original reports were for `FocusDelaySyncJobService`.

`FocusDelaySyncJobService` was removed in PR https://github.com/OneSignal/OneSignal-Android-SDK/pull/1515 and was released in [4.6.6](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/4.6.6).

The followup comment is about an ANR related to `JobScheduler` and unrelated to the original issue. I will post that stack trace in the related issue https://github.com/OneSignal/OneSignal-Android-SDK/issues/1315 to keep track.
